### ISSUE TITLE
Backup strategy for Horcrux

### ISSUE THREAD

**Issue by IDerr** [strangelove-ventures/horcrux#143]
Hello,

I was wondering if there's a defined plan about the backup strategy for sentry / cosigner nodes.

For the sentry node, it seems to be pretty straight forward, we can just stop one of them, make our backup and restart it

What about the cosigner nodes, do we need to backup something on them ?

Thanks a lot for your work!

**Comment by agouin** [comment]
Hello! There are a few things you may want to back up:
- Original `priv_validator_key.json`. As long as you have the original key file, you can re-shard to a new horcrux cluster if necessary. With this, it's unnecessary to back up the `share.json` key shards.
- `priv_validator_state_{chain_id}.json` files - These are useful in case of a validator migration to ensure that the new validator will not sign blocks that have already been signed by the old validator. If you lose these files:
- make sure the old validator is destroyed so there is no chance it will resume.
- make sure the chain is producing blocks (not halted)
- make sure you are missing blocks by checking an explorer and/or other monitoring utility.
If all of these conditions are true, set the state to the latest block height with `horcrux state set [height]` so that there's no chance horcrux will sign a block below a height that you know you have not signed. This is a safe way to restore the state file to a new height.

So to summarize, the private ed25519 key is the only essential thing to back up, but you may decide to back up state files too. The remaining files are ephemeral such as the raft directory and do not need to be backed up.

**Comment by IDerr** [comment]
Perfect, thanks for the quick and precise answer!
